### PR TITLE
Remove build controller

### PR DESCRIPTION
**Pull request by novotnyjakub**
This PR moves the last remaining bit to the ProcessController and removes wicked + BuildController altogether. 

I had to upgrade `capybara-webkit` to 1.7 and it now raises a deprecation warning if compiled with Qt4, so I recommend to compiling it with Qt5 (a qt5 package with homebrew). 

We should bare in mind that the ProcessController now is quite heavy and even has the Rubocop warning for ClassLength supressed. So the next step is to think about what the structure should be and start improving it. But at least we have complete control over the process.

### CODE REVIEW

**Comment by colinbruce** [comment]
Wow! :+1:

File: spec/support/feature_helpers.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module FeatureHelpers
+  def start_new_application
+    visit '/'
+    click_link 'Start now'
+  end
+end
+
+RSpec.configure do |c|
+  c.include FeatureHelpers, type: :feature
+end
```
Comment (dotemacs) [inline spec/support/feature_helpers.rb]: What do we gain with this change?

Why didn't **visit applications_new_path** work?

**Comment by dotemacs** [comment]
What @colinbruce said.

File: spec/support/feature_helpers.rb (hunk 1 above)
Comment (novotnyjakub) [inline spec/support/feature_helpers.rb]: `applications_new_path` isn't a thing any more. It used to be a GET route, which created a new application and redirected to the build. I've made the initial button a POST, which routes to `/applications` and that creates + redirects. 

That means we're not creating objects in a GET action any more, which is correct from RESTful perspective. But we therefore can't simply visit a page to create a new `Application`.

Comment (dotemacs) [inline spec/support/feature_helpers.rb]: Ok, I'll have to look at this properly tomorrow

**Comment by dotemacs** [comment]
Merged master into this, but didn't want to run:

```
git checkout master
git merge --no-ff remove-build-controller
git push origin master
```

just yet, as I wanted to see how it plays out on TravisCI.

**Comment by dotemacs** [comment]
@novotnyjakub ^

**Comment by novotnyjakub** [comment]
Cheers! Just ran through it quickly and it looks good, so if CI passes, we should be good to merge.

**Comment by dotemacs** [comment]
@novotnyjakub we should update **.travis.yml** with the latest qt5 package on debian/ubuntu, so that we suppress the following errors:

```
...WARNING: The next major version of capybara-webkit will require at least version 5.0 of Qt. You're using version 4.8.1.
...
```

**Comment by dotemacs** [comment]
@novotnyjakub thanks, just please double check 6e8b784

**Comment by novotnyjakub** [comment]
Looks good to me.

**Comment by dotemacs** [comment]
GREEN!

**Comment by dotemacs** [comment]
Cool, will merge.

**Pull request merged**